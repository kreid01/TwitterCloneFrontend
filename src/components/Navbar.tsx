import React from 'react';
import { NavLink } from 'react-router-dom'
import { FaHouseUser, FaEnvelope, FaSearch, FaBell, FaBookmark, FaList, FaUser, FaTwitter, FaPenNib } from 'react-icons/fa'
import { IconBaseProps } from 'react-icons';


export default function Navbar() {

    return (
        <header>
            <nav className='fixex w-80 h-screen flex flex-col bg-white shadow-lg'>
                <div className='right-20'>
                <SideBarIcon 
                icon={<FaTwitter size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaHouseUser size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaSearch size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaBell size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaEnvelope size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaBookmark size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaList size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaUser size='30'/>}></SideBarIcon>
                <SideBarIcon 
                icon={<FaPenNib size='30'/>}></SideBarIcon>
                </div>
            </nav>
        </header>
    )
}

const SideBarIcon = ({ icon }: {icon: IconBaseProps})  => (
    <div className='sidebar-icon'>
        <>{icon}</>
    </div>
)