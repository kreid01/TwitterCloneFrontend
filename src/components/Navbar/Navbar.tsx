import React from 'react';
import { NavLink } from 'react-router-dom'
import { FaHouseUser, FaEnvelope, FaSearch, FaBell, FaBookmark, FaList, FaUser, FaTwitter, FaPenNib } from 'react-icons/fa'
import { IconBaseProps } from 'react-icons';


export const Navbar: React.FC = () => {

    return (
        <header data-testid= "nav">
            <nav className='fixed w-20 h-screen flex flex-col bg-white shadow-lg'>
                <NavLink  end to='/'>
                <NavbarIcon 
                icon={<FaTwitter size='30'/>}></NavbarIcon>
                </NavLink>
                <NavLink end to='/'>
                <NavbarIcon 
                icon={<FaHouseUser size='30'/>}></NavbarIcon>
                </NavLink>
                <NavbarIcon 
                icon={<FaSearch size='30'/>}></NavbarIcon>
                <NavbarIcon 
                icon={<FaBell size='30'/>}></NavbarIcon>
                <NavbarIcon 
                icon={<FaEnvelope size='30'/>}></NavbarIcon>
                <NavbarIcon 
                icon={<FaBookmark size='30'/>}></NavbarIcon>
                <NavbarIcon 
                icon={<FaList size='30'/>}></NavbarIcon>
                <NavbarIcon 
                icon={<FaUser size='30'/>}></NavbarIcon>
                <NavbarIcon 
                icon={<FaPenNib size='30'/>}></NavbarIcon>
            </nav>
        </header>
    )
}

export const NavbarIcon = ({ icon }: {icon: IconBaseProps})  => (
    <div className='sidebar-icon'>
        <>{icon}</>
    </div>
)