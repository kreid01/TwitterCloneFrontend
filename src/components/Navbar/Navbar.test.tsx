/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { Navbar } from "./Navbar";

import { screen, render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter, NavLink } from "react-router-dom";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("matched snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
});

it("end to path functions", () => {
  render(
    <BrowserRouter>
      <NavLink end to="/home">
        Home
      </NavLink>
    </BrowserRouter>
  );
  fireEvent.click(screen.getByText("Home"));
  expect(window.location.pathname).toBe("/home");
});
