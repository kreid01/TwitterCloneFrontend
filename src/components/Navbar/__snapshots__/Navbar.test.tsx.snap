// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<header
  data-testid="nav"
>
  <nav
    className="fixed w-20 h-screen flex flex-col bg-white shadow-lg"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <div
        className="sidebar-icon"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          />
        </svg>
      </div>
    </a>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <div
        className="sidebar-icon"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 576 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M570.69,236.27,512,184.44V48a16,16,0,0,0-16-16H432a16,16,0,0,0-16,16V99.67L314.78,10.3C308.5,4.61,296.53,0,288,0s-20.46,4.61-26.74,10.3l-256,226A18.27,18.27,0,0,0,0,248.2a18.64,18.64,0,0,0,4.09,10.71L25.5,282.7a21.14,21.14,0,0,0,12,5.3,21.67,21.67,0,0,0,10.69-4.11l15.9-14V480a32,32,0,0,0,32,32H480a32,32,0,0,0,32-32V269.88l15.91,14A21.94,21.94,0,0,0,538.63,288a20.89,20.89,0,0,0,11.87-5.31l21.41-23.81A21.64,21.64,0,0,0,576,248.19,21,21,0,0,0,570.69,236.27ZM288,176a64,64,0,1,1-64,64A64,64,0,0,1,288,176ZM400,448H176a16,16,0,0,1-16-16,96,96,0,0,1,96-96h64a96,96,0,0,1,96,96A16,16,0,0,1,400,448Z"
          />
        </svg>
      </div>
    </a>
    <div
      className="sidebar-icon"
    >
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </div>
    <div
      className="sidebar-icon"
    >
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
        />
      </svg>
    </div>
    <div
      className="sidebar-icon"
    >
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
        />
      </svg>
    </div>
    <div
      className="sidebar-icon"
    >
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
        />
      </svg>
    </div>
    <div
      className="sidebar-icon"
    >
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
        />
      </svg>
    </div>
    <a
      className=""
      href="/profile"
      onClick={[Function]}
    >
      <div
        className="sidebar-icon"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          />
        </svg>
      </div>
    </a>
    <div
      className="sidebar-icon"
    >
      <svg
        fill="currentColor"
        height="30"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M136.6 138.79a64.003 64.003 0 0 0-43.31 41.35L0 460l14.69 14.69L164.8 324.58c-2.99-6.26-4.8-13.18-4.8-20.58 0-26.51 21.49-48 48-48s48 21.49 48 48-21.49 48-48 48c-7.4 0-14.32-1.81-20.58-4.8L37.31 497.31 52 512l279.86-93.29a64.003 64.003 0 0 0 41.35-43.31L416 224 288 96l-151.4 42.79zm361.34-64.62l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.75 18.75-49.15 0-67.91z"
        />
      </svg>
    </div>
  </nav>
</header>
`;
